### DISCUSSION TITLE
Add support for ED052TC4 (5.2" 720×1280)

### DISCUSSION THREAD

**Discussion by jonasschnelli** [vroland/epdiy#398] [General]
`ED052TC4` is a neat and affordable (10-20$) 5.2" high resolution screen with 720×1280 (WXGA) with 283 PPI similar to the one used in the LilyGo T5 but with higher resolution.

It uses a 50pin `WP27D-P050VA3` connector (see images).
Not sure if an adaptor board for V7 would be possible.

<img width="338" alt="Bildschirmfoto 2024-11-04 um 20 43 21" src="https://github.com/user-attachments/assets/b1590839-a5da-4692-8634-1ae65fd4960c">
<img width="332" alt="Bildschirmfoto 2024-11-04 um 20 45 02" src="https://github.com/user-attachments/assets/e218c986-89d3-44db-b3b2-882204eba117">

https://www.panelook.com/ED052TC4_E_Ink_5.2_EPD_parameter_48664.html

**Comment by martinberlin** [comment]
@jonasschnelli please give some feedback in the next weeks about what is the plan with this display.
Requests of displays that are not supported need some effort of the person wanting to make this work.

**Comment by jonasschnelli** [comment]
@martinberlin My skills in PCB/electronics are limited but I'm happy to contribute (provide screen or work on the firmware part)

**Comment by martinberlin** [comment]
I ordered one of this displays already. But like I mentioned there is existing hardware to try this (That is not super expensive) so you might also get it and try it out first. Just to check if this behaves the way you want. 
ATENTION: This display seems to have a similar connector to Lilygo ED052TC2 but is not the same!
Check datasheet, the connector seems to be https://jlcpcb.com/partdetail/JaeElectronics-WP27D_P050VA3R15000/C3647224

My skills on electronics are also limited. But I just want to give the message that everyone wanting to drive a new display needs to put his effort and time on it to make it happen. As you know there are hundreds of different panels out there and it's impossible to cover all of them even having the advantage that most parallel have the same driving principle (But as you know, different connectors, timings and so on)

**Comment by martinberlin** [comment]
Hello @jonasschnelli  if you get it earlier than me can you please post a Zoomed photo of the connector?
The datasheet is here https://www.data-modul.com/sites/default/files/products/VB3300-EBA-specification-12053269.pdf
And all pictures are quite blurry so it's hard to see that detail. I'm not sure if [C3647224](https://jlcpcb.com/partdetail/JaeElectronics-WP27D_P050VA3R15000/C3647224) is the right female connector where this can be plugged.

**Comment by martinberlin** [comment]
I've designed an adapter for this, still needs time, since I've to wait till display comes and test it. If it works I will share it in the hardware repository
Adapter is ready for a quick review https://github.com/martinberlin/epdiy-hardware-versions/tree/main/adapters/40_to_WP27D-P050VA3
![ED052TC4](https://github.com/user-attachments/assets/a05b5632-e05c-490a-8332-57db1fa01e9b)
@jonasschnelli please be so kind to take a look and follow pins, just to see if I made any mistake. Some of this parts are ordered, at the end made a mistake with the part, and solved it checking in another BOM. Right one is:
https://jlcpcb.com/partdetail/JaeElectronics-WP27D_S050VA3R15000/C3640690 Name is almost the same but has an S instead of a P (It's very sad that there is no relations in the DB pointing what is the female and male connector)

**Comment by jonasschnelli** [comment]
![Groß (IMG_5458)](https://github.com/user-attachments/assets/a0ca1320-7767-428b-8411-c79f56400c12)
![Groß (IMG_5459)](https://github.com/user-attachments/assets/83768287-e550-4dc0-b495-4d75637da4ba)

**Comment by jonasschnelli** [comment]
I reviewed the new adapter schema and have some observations (with my limited understanding):

### Unconnected Pins
* **SPI Pins (1.8V PIN5)**: All SPI pins are unconnected. Purpose of flash memory unclear.
* **TP_1V8 (PIN29)**: Not connected - likely negligible since there's no touch panel
* **Thermistor Connections**:
  * TS (PIN37) - Thermistor sensor not connected
  * TS_GND - Thermistor analog ground not connected
  * Question: What's the purpose of these thermistor pins?

### Connector Verification
* Panel uses: JAE WP27D-P050 VA3
* New adapter: WP27D-S050 VA3
* Note: R15000 designation's significance unknown (likely not relevant)

All other connections appear to match specifications.

**Comment by martinberlin** [comment]
P is Plug and S correspond to Receptacle (in PCB)
Waiting for the part to arrive to JLCPCB. My early adapter was completely WRONG!

You can necer trust importing any FPC connectors pin-out schematics or footprint from Easyeda.

**Comment by martinberlin** [comment]
Update here: Adapter works so anyone that would like to try it I’ve two to give away as a free companion of my epdiy v7 board.
Also I’ve prepared a v7 clone that has the 5.2” connector and additional 6” connector and I’ve called it “v7 raw” since has less components (No UART and no IO expander) that will be sent to fabrication first weeks of January 2025. If that works we can have an open source board that can be adapted for any project with this display.

UPDATE: Due to chinese new year the production will be delayed to 1st week of February 2025

**Comment by martinberlin** [comment]
V7 raw had a few errors and I still don’t know why PWRGOOD pin does not get high. But it works. @vroland left a branch: board/v7-raw for you to test it.
If you like the idea I send you a PCB

https://youtu.be/quSVbgvOYXM?si=UyFBScEK1o5a5Do5

**Comment by martinberlin** [comment]
This is already working converting it to a discussion

**Reply by cxw0102** [reply]
Excuse me, I am using the V5 board driver with the ED047tc2 configuration settings, but the displayed image is mirrored left and right. How can I set it to display correctly?
